### PR TITLE
Javascript Mastery Quiz

### PR DESCRIPTION
**Pull request by mccordgh**
rick and morty forever a hundred times forever www.rickandmorty.com forever

### CODE REVIEW

File: quiz.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	//grab the value from index.html txtLineHeight text box
+	var numCheck = document.getElementById("txtLineHeight").value;
+	
+	//Check if line height value or desired character text fields are blank, warn if so. If both are not blank, run rest of the function
+	if (numCheck === "") {
+		
+		alert("Please enter a height!");
+
+} else if (treeOBJ.char == "") {
```
Comment (JoeShep) [inline quiz.js]: Make sure your formatting is clean before submitting a PR. This should be indented to the same depth as the `if`

File: quiz.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		
+		alert("Please enter the desired character!");
+
+} else {
```
Comment (JoeShep) [inline quiz.js]: indent as above

File: quiz.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+	//clear the console so that concurrent trees don't muck up the place
+	console.clear();
+
+	//simplify later code by grabbing linewidth and the desired character in variables
+		var lineWidth = treeOBJ.height * 2 - 1;
```
Comment (JoeShep) [inline quiz.js]: Everything from here to the bottom is indented too far. This should be same depth as line 21

File: index.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+						document.getElementById('btnGrowTree').click();
+						}">
+				
+				Desired line character: <input type="text" name="txtLineChar" id="txtLineChar" value="*" maxlength="1" style="width:50px;height:40px;font-size:20px;margin:50px;text-align:center" onkeypress="if (event.keyCode == 13) { 
+						document.getElementById('btnGrowTree').click();
+						}">
+				
+				<button type="button" name="btnGrowTree" id="btnGrowTree" style="width:100px;height:100px;font-size:20px;margin:50px;text-align:center" onclick="tree({height: parseInt(document.getElementById('txtLineHeight').value), char: document.getElementById('txtLineChar').value});">
+					Grow Your Tree!
+				</button>
```
Comment (JoeShep) [inline index.html:17]: All of the `on...` functionality you have in your markup should be moved to your javascript and attached to the DOM using event listeners. Inline JS is generally frowned upon.

**Review by JoeShep — COMMENTED** [review]
Hey, Matt. This is working as it should. Great work. I would like you to update a few things, though, to bring this up to best practice standards. 

First, see my comment about your inline JS and move it all to the JS file
Second, you need to clean up your formatting in the JS file. It's haphazard and hard to read. 
Finally, you have everything in one giant function. Try to break this up into smaller functions that have a single job to do.

If you need any help making this happen, let me know. 
Do your edits on this same branch and just push up the changes. No need to close the PR and open another.

**Comment by mccordgh** [comment]
not sure if i need to comment, or if it alerts you to the new commit. 👓

**Review by JoeShep — APPROVED** [review]
Matt, this looks great. Nice work. Only minor quibble: The page layout gets wonky in anything but the widest browser view. Glad you've learned Bootstrap for future projects? : )

I like your use of small helper functions in the building of the tree, and how you clear the console for each new tree. 

Keep up the good work! 👍

**Comment by mccordgh** [comment]
Yeaaaa the page layout is wonky. I am glad that we can use bootstrap now. haaha.

**Pull request merged**